pkg/expressions/json_templates.go

```diff
@@ -99,14 +105,36 @@ func evaluateExpressions(collection any, env map[string]any, exprOpts ...expr.Op
 // EvaluateTemplate evaluates a single template string with the provided
 // environment. Note that a single template string can contain multiple
 // expressions.
+//
+// Standard delimiters are ${{ and }}. When expressions need to contain }}
+// (e.g., nested objects or Go template syntax), alternative delimiters $~~ and ~~
+// can be used instead. The choice of
+// delimiter determines which sequences can appear inside the expression:
+//   - If you start with ${{, then }} cannot appear in the expression, but $~~ can
+//   - If you start with $~~, then ~~ cannot appear in the expression, but ${{ can
 func EvaluateTemplate(template string, env map[string]any, exprOpts ...expr.Option) (any, error) {
-	if !strings.Contains(template, "${{") {
+	// Determine which delimiter pair to use by checking which start tag appears first
+	startTag := "${{"
+	endTag := "}}"
```

Comment:
These should be constants (and used throughout instead of individual string redeclarations).